### PR TITLE
🤖 Sentinel: Kill temporal logic mutants in TimeRange and time utils

### PR DESCRIPTION
**Pull request by madmax983**
This PR targets surviving mutants identified in `src/core/temporal.rs` by `cargo-mutants`.

**🧬 Mutants Found:** Dozens of surviving mutants around `TimeRange` construction boundaries, default values, mathematical drift (`*`, `%` replacing `/`), and explicit boundary overlaps (`<` vs `<=`).
**🎯 Tests Added:** 
- `test_sentinel_timerange_from_exact`
- `test_sentinel_timerange_at_exact`
- `test_sentinel_time_to_secs`
- `test_sentinel_time_to_millis`
- `test_sentinel_timerange_is_closed`
- `test_sentinel_timerange_contains`
**⚠️ Suspected Bugs:** None. These were strictly weak assertions/missing explicit edge-case tests.
**📊 Kill Rate:** Improved coverage and precision of `TimeRange` and `HybridTimestamp` boundary conditions.
**🔗 Havoc Interaction:** None immediately identified. Havoc proptests did not catch these specific arithmetic swaps or boundary default edge-cases.

---
*PR created automatically by Jules for task [7931569226897846172](https://jules.google.com/task/7931569226897846172) started by @madmax983*

### CODE REVIEW

File: .jules/sentinel.md [hunk 1]
```diff
@@ -131,3 +131,9 @@
...
+**Module:** `src/core/temporal.rs`
+**Summary:** Mutants regarding strict bounds on `MAX_VALID_TIMESTAMP` (`>` mutated to `>=` or `==`) within `TimeRange::from` and `TimeRange::at` survived, alongside strict math operators (`*`, `/`, `%`) inside `time::to_secs`, `time::to_millis`. Specific bounding checks involving exclusive interval boundaries in `contains` were weakly asserted allowing `>` to mutate into `>=`.
+**Diagnosis:** MISSING_TEST / WEAK_TEST - Existing tests tested positive path boundary logic well, but neglected specific maximum boundary exact values (`MAX_VALID_TIMESTAMP`), exact conversion validation that strictly broke if `/` turned into `%`, exact exclusion of boundaries inside interval intersections, and robust exact type-checks (like testing that `TimeRange::at()` does not yield an empty string or default timestamp).
+**Kill Shot:** Appended explicit exact boundary tests `test_sentinel_timerange_from_exact`, `test_sentinel_timerange_at_exact`, `test_sentinel_time_to_secs`, `test_sentinel_time_to_millis`, `test_sentinel_timerange_is_closed`, and `test_sentinel_timerange_contains` directly to `src/core/temporal.rs` in a `sentinel_temporal_tests` module.
```
[the remark this answers is not in the document]
Comment (madmax983) [reply .jules/sentinel.md]: > medium
> 
> This new entry appears to duplicate a significant amount of information from the entry on lines 131-133. For instance, both entries mention mutants related to `TimeRange::from`, `TimeRange::at`, `time::to_secs`, `time::to_millis`, and `contains`. To maintain clarity in this tracking file, it would be beneficial to consolidate these entries. If this new entry is intended to supersede the previous one, consider replacing it entirely to avoid redundancy.

@jules
